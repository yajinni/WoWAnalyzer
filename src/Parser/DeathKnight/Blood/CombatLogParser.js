import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
import HealingDone from 'Parser/Core/Modules/HealingDone';
import DamageDone from 'Parser/Core/Modules/DamageDone';
import DamageTaken from 'Parser/Core/Modules/DamageTaken';

import Abilities from './Modules/Abilities';
import Channeling from './Modules/Core/Channeling';
import GlobalCooldown from './Modules/Core/GlobalCooldown';

// Features
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';
import BloodPlagueUptime from './Modules/Features/BloodPlagueUptime';
import CrimsonScourge from './Modules/Features/CrimsonScourge';
import BlooddrinkerTicks from './Modules/Features/BlooddrinkerTicks';
import UnendingThirstTracker from './Modules/Features/UnendingThirstTracker';
import Checklist from './Modules/Features/Checklist';
import MarrowrendUsage from './Modules/Features/MarrowrendUsage';
import Souldrinker from './Modules/Features/Souldrinker';
import BoneShield from './Modules/Features/BoneShield';
import DancingRuneWeapon from './Modules/Features/DancingRuneWeapon';
import InitialMarrowrendCast from './Modules/Features/InitialMarrowrendCast';
import DeathStrikeTiming from './Modules/Features/DeathStrikeTiming';

// Resources
import RunicPowerDetails from './Modules/RunicPower/RunicPowerDetails';
import RunicPowerTracker from './Modules/RunicPower/RunicPowerTracker';
import RuneTracker from '../Shared/RuneTracker';
import RuneDetails from '../Shared/RuneDetails';

// Talents
import RedThirst from './Modules/Talents/RedThirst';
import BoneStorm from './Modules/Talents/Bonestorm';
import MarkOfBloodUptime from './Modules/Talents/MarkOfBlood';
import Bloodworms from './Modules/Talents/Bloodworms';
import Hemostasis from './Modules/Talents/Hemostasis';
import Tombstone from './Modules/Talents/Tombstone';

// Items
import T20_2pc from './Modules/Items/T20_2pc';
import T20_4pc from './Modules/Items/T20_4pc';
import T21_2pc from './Modules/Items/T21_2pc';
import T21_4pc from './Modules/Items/T21_4pc';
import SkullflowersHaemostasis from './Modules/Items/SkullflowersHaemostasis';
import ShacklesofBryndaor from './Modules/Items/ShacklesofBryndaor';
import SoulflayersCorruption from './Modules/Items/SoulflayersCorruption';

//Traits
import RelicTraits from './Modules/Traits/RelicTraits';
import Bonebreaker from './Modules/Traits/Bonebreaker';
import AllConsumingRot from './Modules/Traits/AllConsumingRot';
import Veinrender from './Modules/Traits/Veinrender';
import Coagulopathy from './Modules/Traits/Coagulopathy';


class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core Statistics
    damageTaken: [DamageTaken, { showStatistic: true }],
    damageDone: [DamageDone, { showStatistic: true }],
    healingDone: [HealingDone, { showStatistic: true }],

    // DeathKnight Core
    abilities: Abilities,
    channeling: Channeling,
    globalCooldown: GlobalCooldown,

    // Features
    alwaysBeCasting: AlwaysBeCasting,
    cooldownThroughputTracker: CooldownThroughputTracker,
    crimsonScourge: CrimsonScourge,
    dancingRuneWeapon: DancingRuneWeapon,
    initialMarrowrendCast: InitialMarrowrendCast,
    blooddrinkerTicks: BlooddrinkerTicks,
    unendingThirstTracker:UnendingThirstTracker,
    checklist: Checklist,
    deathStrikeTiming: DeathStrikeTiming,
    marrowrendUsage: MarrowrendUsage,
    souldrinker: Souldrinker,
    boneShield: BoneShield,

    // DOT
    bloodplagueUptime: BloodPlagueUptime,

    // Resources
    runicPowerTracker: RunicPowerTracker,
    runicPowerDetails: RunicPowerDetails,
    runeTracker: RuneTracker,
    runeDetails: RuneDetails,

    // Talents
    redThirst: RedThirst,
    boneStorm: BoneStorm,
    markOfBloodUptime: MarkOfBloodUptime,
    bloodworms: Bloodworms,
    hemostasis: Hemostasis,
    tombstone: Tombstone,

    // Traits
    RelicTraits: RelicTraits,
    bonebreaker: Bonebreaker,
    allConsumingRot: AllConsumingRot,
    veinrender: Veinrender,
    coagulopathy: Coagulopathy,

    // Items:
    t20_2pc: T20_2pc,
    t20_4pc: T20_4pc,
    t21_2pc: T21_2pc,
    t21_4pc: T21_4pc,
    skullflowersHaemostasis: SkullflowersHaemostasis,
    shacklesofBryndaor:ShacklesofBryndaor,
    soulflayersCorruption:SoulflayersCorruption,
  };
}

export default CombatLogParser;
