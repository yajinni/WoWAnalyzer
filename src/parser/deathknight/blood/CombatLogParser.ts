import CoreCombatLogParser from 'parser/core/CombatLogParser';

import Abilities from './modules/Abilities';
import Channeling from './modules/core/Channeling';
import GlobalCooldown from './modules/core/GlobalCooldown';

// Features
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import CooldownThroughputTracker from './modules/features/CooldownThroughputTracker';
import BloodPlagueUptime from './modules/features/BloodPlagueUptime';
import CrimsonScourge from './modules/features/CrimsonScourge';
import BlooddrinkerTicks from './modules/features/BlooddrinkerTicks';
import Checklist from './modules/features/checklist/Module';
import MarrowrendUsage from './modules/features/MarrowrendUsage';
import BoneShield from './modules/features/BoneShield';
import DancingRuneWeapon from './modules/features/DancingRuneWeapon';
import InitialMarrowrendCast from './modules/features/InitialMarrowrendCast';
import DeathStrikeTiming from './modules/features/DeathStrikeTiming';
import BoneShieldTimesByStacks from './modules/features/BoneShieldTimesByStacks';
import DeathsCaress from './modules/core/DeathsCaress';
import MitigationCheck from './modules/features/MitigationCheck';
import Ossuary from './modules/features/Ossuary';

// Resources
import RunicPowerDetails from './modules/runicpower/RunicPowerDetails';
import RunicPowerTracker from './modules/runicpower/RunicPowerTracker';
import RuneTracker from '../shared/RuneTracker';
import RuneDetails from '../shared/RuneDetails';

// Talents
import RedThirst from './modules/talents/RedThirst';
import BoneStorm from './modules/talents/Bonestorm';
import MarkOfBlood from './modules/talents/MarkOfBlood';
import Hemostasis from './modules/talents/Hemostasis';
import FoulBulwark from './modules/talents/FoulBulwark';
import Heartbreaker from './modules/talents/Heartbreaker';
import Bloodworms from './modules/talents/Bloodworms';
import Tombstone from './modules/talents/Tombstone';
import Voracious from './modules/talents/Voracious';
import RapidDecomposition from './modules/talents/RapidDecomposition';
import WillOfTheNecropolis from './modules/talents/WillOfTheNecropolis';
import Consumption from './modules/talents/Consumption';
import RelishInBlood from './modules/talents/RelishInBlood';

// Runes
import RuneForgeChecker from './modules/core/RuneForgeChecker';
import RuneOfTheFallenCrusader from '../shared/runeforges/RuneOfTheFallenCrusader';
import RuneOfHysteria from '../shared/runeforges/RuneOfHysteria';

// Legendaries
import BrynadaorsMight from './modules/items/BrynadaorsMight';
import Superstrain from '../shared/items/Superstrain';

// Covenants
import SwarmingMist from '../shared/covenants/SwarmingMist';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core Statistics
    mitigationCheck: MitigationCheck,

    // DeathKnight Core
    abilities: Abilities,
    channeling: Channeling,
    globalCooldown: GlobalCooldown,

    // Features
    alwaysBeCasting: AlwaysBeCasting,
    cooldownThroughputTracker: CooldownThroughputTracker,
    crimsonScourge: CrimsonScourge,
    dancingRuneWeapon: DancingRuneWeapon,
    initialMarrowrendCast: InitialMarrowrendCast,
    blooddrinkerTicks: BlooddrinkerTicks,
    checklist: Checklist,
    deathStrikeTiming: DeathStrikeTiming,
    marrowrendUsage: MarrowrendUsage,
    boneShield: BoneShield,
    boneShieldTimesByStacks: BoneShieldTimesByStacks,
    deathsCaress: DeathsCaress,

    // DOT
    bloodplagueUptime: BloodPlagueUptime,

    // Resources
    runicPowerTracker: RunicPowerTracker,
    runicPowerDetails: RunicPowerDetails,
    runeTracker: RuneTracker,
    runeDetails: RuneDetails,

    // Talents
    redThirst: RedThirst,
    boneStorm: BoneStorm,
    markOfBlood: MarkOfBlood,
    hemostasis: Hemostasis,
    foulBulwark: FoulBulwark,
    heartbreaker: Heartbreaker,
    bloodworms: Bloodworms,
    tombstone: Tombstone,
    voracious: Voracious,
    rapidDecomposition: RapidDecomposition,
    willOfTheNecropolis: WillOfTheNecropolis,
    ossuary: Ossuary,
    consumption: Consumption,
    relishInBlood: RelishInBlood,

    // Runes
    runeForgeChecker: RuneForgeChecker,
    runeOfTheFallenCrusader: RuneOfTheFallenCrusader,
    runeOfHysteria: RuneOfHysteria,

    // Legendaries
    brynadaorsMight: BrynadaorsMight,
    superStrain: Superstrain,

    // Covenants
    swarmingMist: SwarmingMist
  };
}

export default CombatLogParser;
