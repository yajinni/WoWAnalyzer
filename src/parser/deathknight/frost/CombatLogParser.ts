import CoreCombatLogParser from 'parser/core/CombatLogParser';
import CooldownThroughputTracker from './modules/features/CooldownThroughputTracker';
import Abilities from './modules/Abilities';
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import SpellUsable from './modules/features/SpellUsable';
import Checklist from './modules/features/checklist/Module';

import FrostFeverUptime from './modules/features/FrostFeverUptime';
import RimeEfficiency from './modules/features/RimeEfficiency';
import HardHowlingBlastCasts from './modules/features/HardHowlingBlastCasts';
import KillingMachineEfficiency from './modules/features/KillingMachine';
import BreathOfSindragosa from './modules/features/BreathOfSindragosa';

import RuneTracker from './modules/features/RuneTracker';
import RuneDetails from '../shared/RuneDetails';
import RunicPowerDetails from './modules/runicpower/RunicPowerDetails';
import RunicPowerTracker from './modules/runicpower/RunicPowerTracker';

import GatheringStorm from './modules/talents/GatheringStorm';
import Frostscythe from './modules/talents/Frostscythe';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core
    cooldownThroughputTracker: CooldownThroughputTracker,
    abilities: Abilities,
    alwaysBeCasting: AlwaysBeCasting,
    spellUsable: SpellUsable,
    checklist: Checklist,

    // Features
    HardHowlingBlastCasts: HardHowlingBlastCasts,
    frostfeverUptime: FrostFeverUptime,
    rimeEfficiency: RimeEfficiency,
    killingMachineEfficiency: KillingMachineEfficiency,
    breathofSindragoa: BreathOfSindragosa,

    //resource tracker
    runeTracker: RuneTracker,
    runeDetails: RuneDetails,
    runicPowerDetails: RunicPowerDetails,
    runicPowerTracker: RunicPowerTracker,

    //talents
    gatheringStorm: GatheringStorm,
    frostscythe: Frostscythe,
  };
}

export default CombatLogParser;
