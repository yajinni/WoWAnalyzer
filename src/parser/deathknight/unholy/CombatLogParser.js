import CoreCombatLogParser from 'parser/core/CombatLogParser';

import Abilities from './modules/features/Abilities';
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import CooldownThroughputTracker from './modules/features/CooldownThroughputTracker';
import VirulentPlagueUptime from './modules/features/VirulentPlagueUptime';
import FesteringStrike from './modules/features/FesteringStrike';
import Checklist from './modules/features/checklist/Module';
import ScourgeStrikeEfficiency from './modules/features/ScourgeStrikeEfficiency';
import Apocalypse from './modules/features/Apocalypse';
import VirulentPlagueEfficiency from './modules/features/VirulentPlagueEfficiency';


import RunicPowerDetails from './modules/runicpower/RunicPowerDetails';
import RunicPowerTracker from './modules/runicpower/RunicPowerTracker';

import RuneTracker from './modules/features/RuneTracker';
import RuneDetails from '../shared/RuneDetails';

//Azerite Traits
import BoneSpikeGraveyard from '../shared/spells/azeritetraits/BoneSpikeGraveyard';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Features
    abilities: Abilities,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownThroughputTracker: CooldownThroughputTracker,
    virulentPlagueUptime: VirulentPlagueUptime,
    festeringStrike: FesteringStrike,
    checklist: Checklist,
    scourgeStrikeEfficiency: ScourgeStrikeEfficiency,
  	apocalypse: Apocalypse,
	  virulentPlagueEfficiency: VirulentPlagueEfficiency,

    // RunicPower
    runicPowerTracker: RunicPowerTracker,
    runicPowerDetails: RunicPowerDetails,

    // talents

    //RuneTracker
    runeTracker: RuneTracker,
    runeDetails: RuneDetails,

    //AzeriteTraits
    boneSpikeGraveyard: BoneSpikeGraveyard,
  };
}

export default CombatLogParser;
