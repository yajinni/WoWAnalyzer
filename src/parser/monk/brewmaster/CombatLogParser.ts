import CoreCombatLogParser from 'parser/core/CombatLogParser';
import ArcaneTorrent from 'parser/shared/modules/racials/bloodelf/ArcaneTorrent';

// Core
import HealingDone from './modules/core/HealingDone';
import DamageTaken from './modules/core/DamageTaken';
import HealingReceived from './modules/core/HealingReceived';
import Stagger from './modules/core/Stagger';
import BrewCDR from './modules/core/BrewCDR';
import SharedBrews from './modules/core/SharedBrews';
import StaggerFabricator from './modules/core/StaggerFabricator';
import GlobalCooldown from './modules/core/GlobalCooldown';
import Channeling from './modules/core/Channeling';
import Checklist from './modules/core/Checklist/Module';
// Spells
import PurifyingBrew from './modules/spells/PurifyingBrew';
import CelestialBrew from './modules/spells/CelestialBrew';
import BlackoutCombo from './modules/spells/BlackoutCombo';
import KegSmash from './modules/spells/KegSmash';
import TigerPalm from './modules/spells/TigerPalm';
import RushingJadeWind from './modules/spells/RushingJadeWind';
import BreathOfFire from './modules/spells/BreathOfFire';
import BlackOxBrew from './modules/spells/BlackOxBrew';
import HighTolerance from './modules/spells/HighTolerance';
import CelestialFortune from './modules/spells/CelestialFortune';
import GiftOfTheOxStat from './modules/spells/GiftOfTheOx';
import Shuffle from './modules/spells/Shuffle';
import TouchOfDeath from '../shared/modules/spells/TouchOfDeath';
// Features
import Abilities from './modules/Abilities';
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import StaggerPoolGraph from './modules/features/StaggerPoolGraph';

// Items
// normalizers
import GiftOfTheOx from './normalizers/GiftOfTheOx';
import ExpelHarmNorm from './normalizers/ExpelHarm';

// Covenants
import FallenOrder from '../shared/modules/covenants/FallenOrder';

// Conduits
import ScaldingBrew from './modules/shadowlands/conduits/ScaldingBrew';
import EvasiveStride from './modules/shadowlands/conduits/EvasiveStride';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core
    healingDone: HealingDone,
    healingReceived: HealingReceived,
    damageTaken: DamageTaken,
    stagger: Stagger,
    staggerFabricator: StaggerFabricator,
    brewCdr: BrewCDR,
    brews: SharedBrews,
    channeling: Channeling,
    globalCooldown: GlobalCooldown,
    // There's no throughput benefit from casting Arcane Torrent on cooldown
    arcaneTorrent: [ArcaneTorrent, { castEfficiency: null }] as const,
    checklist: Checklist,

    // Features
    alwaysBeCasting: AlwaysBeCasting,
    abilities: Abilities,
    staggerPoolGraph: StaggerPoolGraph,

    // Spells
    purifyingBrew: PurifyingBrew,
    celestialBrew: CelestialBrew,
    blackoutCombo: BlackoutCombo,
    kegSmash: KegSmash,
    tigerPalm: TigerPalm,
    rjw: RushingJadeWind,
    bof: BreathOfFire,
    bob: BlackOxBrew,
    highTolerance: HighTolerance,
    cf: CelestialFortune,
    gotox: GiftOfTheOxStat,
    shuffle: Shuffle,
    touchOfDeath: TouchOfDeath,

    // Items

    // normalizers
    gotoxNorm: GiftOfTheOx,
    ehNorm: ExpelHarmNorm,

    // Covenants
    fallenOrder: FallenOrder,

    // Conduits
    scaldingBrew: ScaldingBrew,
    evasiveStride: EvasiveStride,
  };
}

export default CombatLogParser;
