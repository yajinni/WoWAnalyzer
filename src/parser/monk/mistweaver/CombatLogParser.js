/* TODO: BfA Edition!
 * Rising Mist - Poor use suggestions
 * Upwelling - Additional healing added from channel, missed healing from channel?
 * Mana Tea vs SotC - Potentially compare common output of each talent.
 *    Suggest using one over the other?
 * Vivify or REM - Missed Vivify healing from less than 2 REMs out
 * Azerite Bonus Placeholders
 */
import CoreCombatLogParser from 'parser/core/CombatLogParser';

import LowHealthHealing from 'parser/shared/modules/features/LowHealthHealing';
import ManaLevelChart from 'parser/shared/modules/resources/mana/ManaLevelChart';
import ManaUsageChart from 'parser/shared/modules/resources/mana/ManaUsageChart';
import LucidDreams from 'parser/shared/modules/spells/bfa/essences/LucidDreamsHealers';

import GlobalCooldown from './modules/core/GlobalCooldown';
import CoreChanneling from './modules/core/Channeling';
import HotTracker from './modules/core/HotTracker';
import SpellUsable from './modules/core/SpellUsable';

// Normalizers
import HotApplicationNormalizer from './normalizers/HotApplicationNormalizer';
import HotRemovalNormalizer from './normalizers/HotRemovalNormalizer';

// Features
import Abilities from './modules/features/Abilities';
import CooldownThroughputTracker from './modules/features/CooldownThroughputTracker';
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import EssenceFontMastery from './modules/features/EssenceFontMastery';
import Checklist from './modules/features/Checklist/Module';
import StatValues from './modules/features/StatValues';
import EvmVivCastRatio from './modules/features/EvmVivCastRatio';
import MasteryStats from './modules/features/MasteryStats';
import Buffs from './modules/features/Buffs';

// Spells
import ThunderFocusTea from './modules/spells/ThunderFocusTea';
import EssenceFont from './modules/spells/EssenceFont';
import EnvelopingMists from './modules/spells/EnvelopingMists';
import SoothingMist from './modules/spells/SoothingMist';
import Vivify from './modules/spells/Vivify';
import LifeCocoon from './modules/spells/LifeCocoon';
import SpinningCraneKick from './modules/spells/SpinningCraneKick';
import RenewingMist from './modules/spells/RenewingMist';

// Talents
import JadeSerpentStatue from './modules/talents/JadeSerpentStatue';
import ChiJi from './modules/talents/ChiJi';
import ChiBurst from './modules/talents/ChiBurst';
import ManaTea from './modules/talents/ManaTea';
import RefreshingJadeWind from './modules/talents/RefreshingJadeWind';
import Lifecycles from './modules/talents/Lifecycles';
import SpiritOfTheCrane from './modules/talents/SpiritOfTheCrane';
import RisingMist from './modules/talents/RisingMist';
import AverageTimeBetweenRSKSs from './modules/talents/AverageTimeBetweenRSKs';
import RenewingMistDuringManaTea from './modules/talents/RenewingMistDuringManaTea';
import Tier45Comparison from './modules/talents/Tier45Comparison';
import Upwelling from './modules/talents/Upwelling';

// Azerite Traits
import FontOfLife from './modules/spells/azeritetraits/FontOfLife';
import UpliftedSpirits from './modules/spells/azeritetraits/UpliftedSpirits';
import SecretInfusion from './modules/spells/azeritetraits/SecretInfusion';
import MistyPeaks from './modules/spells/azeritetraits/MistyPeaks';

//essences
import WayOfTheCrane from './modules/spells/essences/WayOfTheCrane';

// Mana Tracker
import MistweaverHealingEfficiencyDetails from './modules/features/MistweaverHealingEfficiencyDetails';
import HealingEfficiencyTracker from './modules/features/MistweaverHealingEfficiencyTracker';
import ManaTracker from '../../core/healingEfficiency/ManaTracker';



import { ABILITIES_AFFECTED_BY_HEALING_INCREASES } from './constants';
import EssenceFontHealingBreakdown from './modules/features/EssenceFontHealingBreakdown';


class CombatLogParser extends CoreCombatLogParser {
  static abilitiesAffectedByHealingIncreases = ABILITIES_AFFECTED_BY_HEALING_INCREASES;

  static specModules = {
    // Normalizer
    hotApplicationNormalizer: HotApplicationNormalizer,
    hotRemovalNormalizer: HotRemovalNormalizer,

    // Core
    lowHealthHealing: LowHealthHealing,
    channeling: CoreChanneling,
    globalCooldown: GlobalCooldown,
    hotTracker: HotTracker,
    spellUsable: SpellUsable,

    // Generic healer things
    manaLevelChart: ManaLevelChart,
    manaUsageChart: ManaUsageChart,

    // Features
    alwaysBeCasting: AlwaysBeCasting,
    abilities: Abilities,
    cooldownThroughputTracker: CooldownThroughputTracker,
    essenceFontMastery: EssenceFontMastery,
    checklist: Checklist,
    statValues: StatValues,
    evmVivCastRatio: EvmVivCastRatio,
    masteryStats: MasteryStats,
    buffs: Buffs,
    essenceFontHealingBreakDown: EssenceFontHealingBreakdown,
    averageTimeBetweenRSKSs: AverageTimeBetweenRSKSs,

    // Spells
    essenceFont: EssenceFont,
    thunderFocusTea: ThunderFocusTea,
    envelopingMists: EnvelopingMists,
    soothingMist: SoothingMist, // Removed as this needs to be reworked with updated Soothing Mist Spell in BfA
    spinningCraneKick: SpinningCraneKick,
    vivify: Vivify,
    renewingMist: RenewingMist,
    lifeCocoon: LifeCocoon,

    // Talents
    chiBurst: ChiBurst,
    chiJi: ChiJi,
    manaTea: ManaTea,
    refreshingJadeWind: RefreshingJadeWind,
    lifecycles: Lifecycles,
    spiritOfTheCrane: SpiritOfTheCrane,
    risingMist: RisingMist,
    jadeSerpentStatue: JadeSerpentStatue,
    renewingMistDuringManaTea: RenewingMistDuringManaTea,
    tier45Comparison: Tier45Comparison,
    upwelling: Upwelling,

    // Azerite Traits
    fontOfLife: FontOfLife,
    upliftedSpirits: UpliftedSpirits,
    secretInfusion: SecretInfusion,
    mistyPeaks: MistyPeaks,

    // Essences
    lucidDreams: LucidDreams,

    // Mana Tab
    manaTracker: ManaTracker,
    hpmDetails: MistweaverHealingEfficiencyDetails,
    hpmTracker: HealingEfficiencyTracker,

    //Essences
    wayOfTheCrane: WayOfTheCrane,
  };
}

export default CombatLogParser;
