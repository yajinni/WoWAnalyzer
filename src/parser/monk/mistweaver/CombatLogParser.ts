import CoreCombatLogParser from 'parser/core/CombatLogParser';

import LowHealthHealing from 'parser/shared/modules/features/LowHealthHealing';
import ManaLevelChart from 'parser/shared/modules/resources/mana/ManaLevelChart';
import ManaUsageChart from 'parser/shared/modules/resources/mana/ManaUsageChart';

import GlobalCooldown from './modules/core/GlobalCooldown';
import CoreChanneling from './modules/core/Channeling';
import HotTrackerMW from './modules/core/HotTrackerMW';

// Normalizers
import HotApplicationNormalizer from './normalizers/HotApplicationNormalizer';
import HotRemovalNormalizer from './normalizers/HotRemovalNormalizer';

// Features
import Abilities from './modules/features/Abilities';
import CooldownThroughputTracker from './modules/features/CooldownThroughputTracker';
import AlwaysBeCasting from './modules/features/AlwaysBeCasting';
import EssenceFontMastery from './modules/features/EssenceFontMastery';
import Checklist from './modules/features/Checklist/Module';
import StatValues from './modules/features/StatValues';
import EvmVivCastRatio from './modules/features/EvmVivCastRatio';
import MasteryStats from './modules/features/MasteryStats';
import Buffs from './modules/features/Buffs';

// Spells
import ThunderFocusTea from './modules/spells/ThunderFocusTea';
import EssenceFont from './modules/spells/EssenceFont';
import EnvelopingMists from './modules/spells/EnvelopingMists';
import SoothingMist from './modules/spells/SoothingMist';
import Vivify from './modules/spells/Vivify';
import LifeCocoon from './modules/spells/LifeCocoon';
import SpinningCraneKick from './modules/spells/SpinningCraneKick';
import RenewingMist from './modules/spells/RenewingMist';
import TouchOfDeath from '../shared/modules/spells/TouchOfDeath';
import InvokeYulon from './modules/spells/InvokeYulon';
import InvokeChiJi from './modules/talents/InvokeChiJi';
import ExpelHarm from './modules/spells/ExpelHarm';
import EnvelopingBreath from './modules/spells/EnvelopingBreath';

// Talents
import JadeSerpentStatue from './modules/talents/JadeSerpentStatue';
import ChiBurst from './modules/talents/ChiBurst';
import ManaTea from './modules/talents/ManaTea';
import RefreshingJadeWind from './modules/talents/RefreshingJadeWind';
import Lifecycles from './modules/talents/Lifecycles';
import SpiritOfTheCrane from './modules/talents/SpiritOfTheCrane';
import RisingMist from './modules/talents/RisingMist';
import AverageTimeBetweenRSKSs from './modules/talents/AverageTimeBetweenRSKs';
import RenewingMistDuringManaTea from './modules/talents/RenewingMistDuringManaTea';
import Tier30Comparison from './modules/talents/Tier30Comparison';
import Upwelling from './modules/talents/Upwelling';

// Mana Tracker
import MistweaverHealingEfficiencyDetails from './modules/features/MistweaverHealingEfficiencyDetails';
import HealingEfficiencyTracker from './modules/features/MistweaverHealingEfficiencyTracker';
import ManaTracker from '../../core/healingEfficiency/ManaTracker';

import { ABILITIES_AFFECTED_BY_HEALING_INCREASES } from './constants';
import EssenceFontHealingBreakdown from './modules/features/EssenceFontHealingBreakdown';

// Covenants
import FallenOrder from '../shared/modules/covenants/FallenOrder';
import FaelineStomp from '../shared/modules/covenants/FaelineStomp';

// Conduits
// Endurance
import GroundingBreath from '../shared/modules/conduits/GroundingBreath';
import HarmDenial from '../shared/modules/conduits/HarmDenial';
import FortifyingIngredients from '../shared/modules/conduits/FortifyingIngredients';

// Potency
import JadeBond from './modules/shadowlands/conduits/JadeBond';
import NourishingChi from './modules/shadowlands/conduits/NourishingChi';
import RisingSunRevival from './modules/shadowlands/conduits/RisingSunRevival';
import ImbuedReflections from '../shared/modules/conduits/ImbuedReflections';

// Legendaries
import TearofMorning from './modules/shadowlands/legendaries/TearofMorning';
import AncientTeachingsoftheMonastery from './modules/shadowlands/legendaries/AncientTeachingsoftheMonastery';
import CloudedFocus from './modules/shadowlands/legendaries/CloudedFocus';

class CombatLogParser extends CoreCombatLogParser {
  static abilitiesAffectedByHealingIncreases = ABILITIES_AFFECTED_BY_HEALING_INCREASES;

  static specModules = {
    // Normalizer
    hotApplicationNormalizer: HotApplicationNormalizer,
    hotRemovalNormalizer: HotRemovalNormalizer,

    // Core
    lowHealthHealing: LowHealthHealing,
    channeling: CoreChanneling,
    globalCooldown: GlobalCooldown,
    hotTrackerMW: HotTrackerMW,

    // Generic healer things
    manaLevelChart: ManaLevelChart,
    manaUsageChart: ManaUsageChart,

    // Features
    alwaysBeCasting: AlwaysBeCasting,
    abilities: Abilities,
    cooldownThroughputTracker: CooldownThroughputTracker,
    essenceFontMastery: EssenceFontMastery,
    checklist: Checklist,
    statValues: StatValues,
    evmVivCastRatio: EvmVivCastRatio,
    masteryStats: MasteryStats,
    buffs: Buffs,
    essenceFontHealingBreakDown: EssenceFontHealingBreakdown,
    averageTimeBetweenRSKSs: AverageTimeBetweenRSKSs,

    // Spells
    essenceFont: EssenceFont,
    thunderFocusTea: ThunderFocusTea,
    envelopingMists: EnvelopingMists,
    soothingMist: SoothingMist, // Removed as this needs to be reworked with updated Soothing Mist Spell in BfA
    spinningCraneKick: SpinningCraneKick,
    vivify: Vivify,
    renewingMist: RenewingMist,
    lifeCocoon: LifeCocoon,
    touchOfDeath: TouchOfDeath,
    invokeYulon: InvokeYulon,
    expelHarm: ExpelHarm,
    envelopingBreath: EnvelopingBreath,

    // Talents
    chiBurst: ChiBurst,
    manaTea: ManaTea,
    refreshingJadeWind: RefreshingJadeWind,
    lifecycles: Lifecycles,
    spiritOfTheCrane: SpiritOfTheCrane,
    risingMist: RisingMist,
    jadeSerpentStatue: JadeSerpentStatue,
    renewingMistDuringManaTea: RenewingMistDuringManaTea,
    tier30Comparison: Tier30Comparison,
    upwelling: Upwelling,
    invokeChiJi: InvokeChiJi,

    // Mana Tab
    manaTracker: ManaTracker,
    hpmDetails: MistweaverHealingEfficiencyDetails,
    hpmTracker: HealingEfficiencyTracker,

    // Covenants
    fallenOrder: FallenOrder,
    faelineStomp: FaelineStomp,

    // Conduits
    // Endurance
    groundingBreath: GroundingBreath,
    harmDenial: HarmDenial,
    fortifyingIngredients: FortifyingIngredients,

    // Potency
    jadeBond: JadeBond,
    nourishingChi: NourishingChi,
    risingSunRevival: RisingSunRevival,
    imbuedReflections: ImbuedReflections,

    // Legendaries
    tearofMorning: TearofMorning,
    ancientTeachingsoftheMonastery: AncientTeachingsoftheMonastery,
    cloudedFocus: CloudedFocus,
  };
}

export default CombatLogParser;
