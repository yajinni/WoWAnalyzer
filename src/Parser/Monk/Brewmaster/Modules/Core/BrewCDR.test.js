// naming & location of this test file is for ease of location, there is
// a web of modules needed for this test so we are just loading up
// everything here
import zlib from 'zlib';
import fs from 'fs';
import CombatLogParser from '../../CombatLogParser'; 
import BrewCDR from './BrewCDR';
import report from '../../test-logs/report_nxzcqJKmYwHbdWMp';

// Disable spammy warnings
console.error = jest.fn();
console.warn = jest.fn();

describe.skip('BrewCDR', () => {
  let events;
  let parser;

  beforeAll(() => {
    const buff = fs.readFileSync(__dirname + "/../../test-logs/eisenpelz_felhounds_h_nxzcqJKmYwHbdWMp.json.gz");
    events = JSON.parse(zlib.unzipSync(buff).toString()).events;
    parser = new CombatLogParser(
      report,
      report.friendlies.find(player => player.name === "Eisenpelz"),
      [],
      report.fights.find(fight => fight.id === 8)
    );
    parser.initialize(events.filter(ev => ev.type === "combatantinfo"));
    parser.parseEvents(events);
  });

  it("should contain an active BrewCDR module", () => {
    expect(parser.findModule(BrewCDR).active).toBe(true);
  });

  it("should report 427.09s of CDR", () => {
    expect(Math.round(parser.findModule(BrewCDR).totalCDR)).toBe(451469);
  });
});
