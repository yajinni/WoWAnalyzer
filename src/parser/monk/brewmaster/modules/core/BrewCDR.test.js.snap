// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrewCDR should match the cdr snapshot 1`] = `0.2630088802883363`;
